### DISCUSSION TITLE
Stripe Webhooks in test mode

### DISCUSSION THREAD

**Discussion by dwarfmondo** [pay-rails/pay#560] [Q&A] [answered]
I have Stripe Webhooks in "Test mode" pointed to our staging environment. We're using the same Stripe account in "Test mode" for local development. This means that data (subscriptions, charges, etc.) from our development environment are pinging our staging environment and returning a 400. I assume Pay is returning the 400 because it can't find the subscription or charge in the staging database.

What's the best practice for handling this? I'd like to see everything working end-to-end in our staging environment before confidently promoting to production. Is there a way to have Pay not return a 400 if the `STRIPE_ENV` is set to something other than production.

Thanks for the help.

**Comment by excid3** [comment, marked as answer]
`400` is only return returned if the Stripe signing secret cannot verify the webhook. Sounds like you don't have the `test` signing secret configured on staging?

https://github.com/pay-rails/pay/blob/master/app/controllers/pay/webhooks/stripe_controller.rb#L11-L13

**Reply by dwarfmondo** [reply]
Ooof. That's embarrassing. I did indeed have the wrong signing secret. Thanks a ton for the help!
